lib/finsta_web/live/post_live/form_component.ex

```diff
@@ -19,7 +24,16 @@ defmodule FinstaWeb.PostLive.FormComponent do
         phx-change="validate"
         phx-submit="save"
       >
+        <%= for image <- @uploads.image.entries do %>
+          <figure>
+            <.live_img_preview entry={image} />
+          </figure>
+
+          <progress value={image.progress} max="100"><%= image.progress %>%</progress>
+        <% end %>
+
         <.input field={@form[:caption]} type="text" label="Caption" />
+        <.live_file_input :if={!@post.image_url} upload={@uploads.image} required />
```

Comment:
This `:if={!@post.image_url}` is to prevent the user from updating the image when editing a post. Only the caption is allowed to be changed.